pyomo/common/collections/_hasher.py

```diff
@@ -0,0 +1,52 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2025
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from collections import defaultdict
+
+
+class _Hasher(defaultdict):
```

Comment:
We talked about this on the dev call, but I think it would be worth removing the `_` -- Sphinx by default will not document private classes, attributes, etc.

Of course, we should also add documentation to this class...